### DISCUSSION TITLE
PAdES

### DISCUSSION THREAD

**Discussion by Ziberoncheg** [MatthiasValvekens/pyHanko#104] [Q&A]
Hey. I need to sign pdf document on PAsES alg (sign inside document. one file with .pdf).

when i use key from DIA (.pfx), i get error: no attribute 'get_signature_mechanism'.

Do you have any sollution for this?

**Comment by MatthiasValvekens** [comment]
Sorry, I'm not sure I understand your question. An attribute error like `get_signature_mechanism` indicates a programming error. From the looks of it, you probably passed an object of a different type where the API expects a `Signer` instance---i.e. the symptoms you're seeing likely have nothing to do with the nature of the key material or whether you're signing in PAdES mode.

Could you post a full stack trace with some sample code, please? That would probably allow us to identify what's going wrong.
